src/main/java/io/sqooba/traildb/TrailDBIterator.java

```diff
@@ -52,6 +52,9 @@ public TrailDBEvent next() {
             public boolean hasNext() {
                 TrailDBEvent next = TrailDBNative.INSTANCE.cursorNext(TrailDBIterator.this.cursor);
                 if (next == null) {
+                    // Auto-close when at the end because it is not intuitive to have to close an
+                    // iterator when at its end in Java.
+                    TrailDBIterator.this.close();
```

Comment:
I will rephrase.
We have to close it only if we are done because then we can not use it anymore, so if we do it in the next method, a subsequent call to next would fail.
I agree that the current state of the iterator is not great. It has been implemented in a manner that it reflects what is done in the C code but at the same time I tried to make it like a Java iterator and the two seem to not work well together.